src/frontend/src/catalog/table_catalog.rs

```diff
@@ -239,15 +239,7 @@ impl TableCatalog {
                 "Use `DROP MATERIALIZED VIEW` to drop a materialized view."
             }
             TableType::Index => "Use `DROP INDEX` to drop an index.",
-            TableType::Table => {
-                // TODO(Yuanxin): Remove this after unsupporting `CREATE MATERIALIZED SOURCE`.
-                // Note(bugen): may make this a method on `TableType` instead.
```

Comment:
What about implementing this directly on `TableType`?

Reply:
Not sure whether it's a good idea to separate `Table` and `TableWithConnector` as two types. cc @st1page

Reply:
Both LGTM. Never mind. 😄